src/room.js

```diff
@@ -0,0 +1,40 @@
+class Room {
+
+  constructor (roomname) {
+    this.roomname = roomname;
+    this.endpoints = {};
+  }
+  addEndpoint (endpointId) {
+    this.endpoints[endpointId] = {
```

Comment:
Still got this issue from before:

Rather than initialising endpoints with some empty strings, we should just have `this.endpoints = {}`

Whereas the `addEndpoint` method should create a new endpointId, like I have done [here](https://github.com/foundersandcoders/Live-Peers/pull/56/files#diff-c8a2bf410c4a6e6dcf012df0da242442R10):
```javascript
addEndpoint (roomId) {
  let endpointId = // function to create a random string
  this.endpoints[endpointId] = {};
}
```

And then we need a few more methods:
+ addUsername
+ addPermissions
+ addCommsId